pkg/logs/internal/launchers/file/launcher.go

```diff
@@ -60,10 +61,20 @@ type Launcher struct {
 }
 
 // NewLauncher returns a new launcher.
-func NewLauncher(tailingLimit int, tailerSleepDuration time.Duration, validatePodContainerID bool, scanPeriod time.Duration) *Launcher {
+func NewLauncher(tailingLimit int, tailerSleepDuration time.Duration, validatePodContainerID bool, scanPeriod time.Duration, wildcardMode string) *Launcher {
+	var prov *fileprovider.FileProvider
+	if wildcardMode == "prioritizeNewest" {
+		prov = fileprovider.NewFileProvider(tailingLimit, fileprovider.WildcardMtime, fileprovider.GlobalSelection)
+	} else if wildcardMode == "simpleAlpha" {
+		prov = fileprovider.NewFileProvider(tailingLimit, fileprovider.WildcardReverseLexicographical, fileprovider.GreedySelection)
```

Comment:
Selection mode and wildcard mode map 1 to 1: do we really need to expose it in the constructor?

Reply:
All combinations of selection mode and wildcard mode are valid and tested. Ie, you can specify `GlobalSelection` with `ReverseLexicographical` sorting.

So from "what is the public API of this component" perspective, it makes sense to me to expose both, even though we only use 2 combinations.

Reply:
[Swapping](https://gist.github.com/vickenty/6b03029eab9d2ac4e5d988bc0cd3f604) `Greedy` and `Global` here does not seem to fail any tests, which means a mistake can creep in. Of course it can be covered by more tests,  but I think it is better to have minimal interfaces: fewer behaviors means less code and tests (which affects amount of future maintenance work required), and less chances to trigger the wrong behavior by mistake.